webrtc/interfaces.html

```diff
@@ -0,0 +1,55 @@
+<!doctype html>
+<html>
+  <head>
+    <title>"WebRTC IDL Tests</title>
+  </head>
+  <body>
+    <script src=/resources/testharness.js></script>
+    <script src=/resources/testharnessreport.js></script>
+    <script src=/resources/WebIDLParser.js></script>
+    <script src=/resources/idlharness.js></script>
+    <script>
+      'use strict';
+
+      function initRTCCertificate() {
+        return RTCPeerConnection.generateCertificate({
+          name: 'RSASSA-PKCS1-v1_5',
+          modulusLength: 2048,
+          publicExponent: new Uint8Array([1, 0, 1]),
+          hash: 'SHA-256'
+        })
+      }
+
+      function doIdlTest(idlText) {
+        const idlArray = new IdlArray();
+
+        idlArray.add_untested_idls('interface EventHandler {};');
+        idlArray.add_untested_idls('interface MediaStreamTrack {};');
+        idlArray.add_idls(idlText);
+
+        // TODO: Add object for all IDL interfaces
+        idlArray.add_objects({
+          'RTCPeerConnection': ['new RTCPeerConnection'],
+          'RTCSessionDescription': ['new RTCSessionDescription({ type: "offer" })'],
+          'RTCIceCandidate': ['new RTCIceCandidate'],
+          'RTCPeerConnectionIceEvent': ['new RTCPeerConnectionIceEvent("ice")'],
+          'RTCPeerConnectionIceErrorEvent': ['new RTCPeerConnectionIceErrorEvent("ice-error", { errorCode: 701 });'],
+        });
+
+        return initRTCCertificate()
+          .then(cert => {
+            window.rtcCert = cert;
```

Comment:
The caveat here is that none of the IDL tests will run if `RTCPeerConnection.generateCertificate` fails. I am looking for a way to catch the promise rejection, report it to testharness, and proceed the rest of IDL tests regardless. The simplest solution without additional code is to modify `promise_test` to return a promise of the test completing.